include/zephyr/min_heap.h

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) 2025 Aerlync Labs Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_MIN_HEAP_H_
+#define ZEPHYR_INCLUDE_MIN_HEAP_H_
+
+#include <zephyr/sys/util.h>
+#include <zephyr/kernel.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief min_heap
+ * @defgroup min_heap Min-heap service
+ * @ingroup os_services
+ * @{
+ */
+
+/**
+ * @brief Invasive node used inside user structs for min-heap support.
+ */
+struct min_heap_node {
+	/** Index within heap array */
+	size_t index;
+};
+
+/**
+ * @brief Comparator function type for min-heap ordering.
+ *
+ * This function compares two heap nodes to establish their relative order.
+ * It must be implemented by the user and provided at min-heap
+ * initialization.
+ *
+ * @param a Pointer to the first node.
+ * @param b Pointer to the second node.
+ *
+ * @return Negative value if @p a is less than @p b,
+ *         positive value if @p a is greater than @p b,
+ *         zero if they are equal.
+ */
+typedef int (*min_heap_cmp_fn)(const struct min_heap_node *a,
+			       const struct min_heap_node *b);
+
+/**
+ * @brief min-heap data structure with user-provided comparator.
+ */
+struct min_heap {
+	/** Pointer to preallocated node pointer array */
+	struct min_heap_node **nodes;
+	/** Maximum number of elements */
+	size_t capacity;
+	/** Current elements count */
+	size_t size;
+	/** Comparator function */
+	min_heap_cmp_fn cmp;
+};
+
+/**
+ * @brief Static initializer macro for min-heap.
+ *
+ * This macro provides a convenient way to statically initialize a min_heap
+ * structure with a given node storage array, capacity,
+ * and comparator function.
+ *
+ * @param storage Preallocated array of node pointers
+ * @param cap Number of elements in the array
+ * @param cmp_func Comparator function
+ */
+#define MIN_HEAP_INIT(storage, cap, cmp_func) \
```

Comment:
Is it possible to add a `_DEFINE` macro maybe along with `_INIT` that would declare a named instance?

I think that is maybe a more typical behaviour for declaration-oriented macros in Zephyr.